optimism/src/lookup.rs

```diff
@@ -201,4 +206,28 @@ impl<F: Field> LookupTable<F> {
                 .collect(),
         }
     }
+
+    fn _table_pad() -> Self {
+        Self {
+            _table: (1..=136)
+                .map(|i| {
+                    let suffix = pad_blocks(i as usize);
+                    Lookup {
+                        mode: LookupMode::Write,
+                        magnitude: F::one(),
+                        table_id: LookupTables::PadLookup,
+                        value: vec![
+                            F::from(i),
+                            F::two_pow(i),
```

Comment:
Why do you need `F::two_pow(i)` here? I assume you need it for the application logic to restrain `y` to be `2^i` for given (`i`,`y`)? And creating a separate table is just more expensive?

Reply:
Exactly, in some constraints to check that the padding is performed at the end of the bytestring I need the value `2^i`. Creating another table just for this would be more expensive (essentially 135 more rows in the circuit). Unlike with the sparse representation, in this case I will be able to access the whole entry at once, no need to look at individual columns. So for that reason, there's no need to include it in a separate table.

IF at some point we needed the powers of two for whatever other reason, then it would make sense to have it separate somewhere else. But for now, it's fine to reuse the same table.